.github/workflows/integration-tests.yml

```diff
@@ -21,11 +21,6 @@ jobs:
         with:
           components: clippy, rustfmt
       
-      - name: Cache dependencies
```

Comment:
Unrelated to main PR intention, but unfortunately, I've to remove this. In the devops GH runners, pulling the cache works at ~1MiB/s, and rust caches are quite big so it's quite slow.

<img width="952" height="524" alt="image" src="https://github.com/user-attachments/assets/61386158-4c72-418c-b6ae-60bb9802e9ee" />